index.js

```diff
@@ -8,10 +8,17 @@ function difference (arrA, arrB) {
     return arrA.filter(a => arrB.indexOf(a) < 0);
 }
 
-function compact (obj) {
+function maybeRemoveEmptyValues (obj, allowEmptyValues) {
+    function allowIfEmpty (key) {
+        return (
+            allowEmptyValues === true ||
+            (Array.isArray(allowEmptyValues) && allowEmptyValues.includes(key))
```

Comment:
Hi @williamdclt, thanks for your contribution!

The downside of this approach is that we're checking if `allowEmptyValues` is an array and also iterating it for every non-empty value in `process.env`. It would be faster to accept a `Set` object as an argument instead of an array, and then check only once if the provided argument was a boolean or a `Set`.

I don't see any downsides of accepting a `Set` instead of an array but then again I haven't thought this fully through. Let me know what you think.